File: cc/array/DetermineColorOfAChessboardSquare.cc
#include <string>

using namespace std;

/**
 * LeetCode 1812. Determine Color of a Chessboard Square
 * https://leetcode.com/problems/determine-color-of-a-chessboard-square/
 */
class DetermineColorOfAChessboardSquare {
public:
    bool squareIsWhite(string coordinates) {
        if (coordinates.size() != 2) {
            return false;
        }
        return ((coordinates[0] - 'a') + (coordinates[1] - '1')) % 2 == 1;
    }
};

int main(int argc, char const *argv[]) {
    // ...
    return 0;
}

/* EOF */

File: java/src/io/zihengcat/leetcode/array/DetermineColorOfAChessboardSquare.java
package io.zihengcat.leetcode.array;

/**
 * LeetCode 1812. Determine Color of a Chessboard Square
 * https://leetcode.com/problems/determine-color-of-a-chessboard-square/
 */
public class DetermineColorOfAChessboardSquare {
    public static void main(String[] args) {
        // ...
    }
    public boolean squareIsWhite(String coordinates) {
        if (coordinates == null || coordinates.length() != 2) {
            return false;
        }
        return ((coordinates.charAt(0) - 'a') + (coordinates.charAt(1) - '1')) % 2 == 1;
    }
}

/* EOF */
